shadow.c

```diff
@@ -117,10 +118,19 @@ static PHP_GSHUTDOWN_FUNCTION(shadow)
 }
 /* }}} */
 
+/* {{{ shadow_deps
+ */
+static const zend_module_dep shadow_deps[] = {
+	ZEND_MOD_OPTIONAL("zip")
```

Comment:
If this is only needed to be able to run `make test`, it should be solved in somewhere in the make file, not in the source code.

Right now, after executing `phpize` in the extension directory, the resulting Makefile.global contains:
```
-d extension_dir=$(top_builddir)/modules/ $(PHP_TEST_SHARED_EXTENSIONS)
```

If we swap extensions, the tests will pass.

Particularly for Travis, we can do this in .travis.yml like [this](https://github.com/sugarcrm/shadow/blob/master/.travis.yml#L15). But ideally, if that's possible we need to patch the Makefile, so even if we run `make test` locally, the extension is loaded the last. This approach will guarantee that tests will pass with whatever set of shared extensions and will be an example of how the server needs to be configured.